app/src/main/java/io/homeassistant/companion/android/sensors/CarSensorManager.kt

```diff
@@ -409,154 +410,144 @@ class CarSensorManager :
     private fun onEnergyAvailable(data: EnergyLevel) {
         val fuelStatus = carValueStatus(data.fuelPercent.status)
         Log.d(TAG, "Received Energy level: $data")
-        if (isEnabled(latestContext, fuelLevel)) {
-            onSensorUpdated(
-                latestContext,
-                fuelLevel.sensor,
-                if (fuelStatus == "success") data.fuelPercent.value!! else STATE_UNKNOWN,
-                fuelLevel.sensor.statelessIcon,
-                mapOf(
-                    "status" to fuelStatus
-                ),
-                forceUpdate = true
-            )
-        }
+        onSensorUpdated(
+            latestContext,
+            fuelLevel.sensor,
+            if (fuelStatus == "success") data.fuelPercent.value!! else STATE_UNKNOWN,
+            fuelLevel.sensor.statelessIcon,
+            mapOf(
+                "status" to fuelStatus
+            ),
+            forceUpdate = true
+        )
         val batteryStatus = carValueStatus(data.batteryPercent.status)
-        if (isEnabled(latestContext, batteryLevel)) {
-            onSensorUpdated(
-                latestContext,
-                batteryLevel.sensor,
-                if (batteryStatus == "success") data.batteryPercent.value!! else STATE_UNKNOWN,
-                batteryLevel.sensor.statelessIcon,
-                mapOf(
-                    "status" to batteryStatus
-                ),
-                forceUpdate = true
-            )
-        }
+        onSensorUpdated(
+            latestContext,
+            batteryLevel.sensor,
+            if (batteryStatus == "success") data.batteryPercent.value!! else STATE_UNKNOWN,
+            batteryLevel.sensor.statelessIcon,
+            mapOf(
+                "status" to batteryStatus
+            ),
+            forceUpdate = true
+        )
         val rangeRemainingStatus = carValueStatus(data.rangeRemainingMeters.status)
-        if (isEnabled(latestContext, rangeRemaining)) {
-            onSensorUpdated(
-                latestContext,
-                rangeRemaining.sensor,
-                if (rangeRemainingStatus == "success") data.rangeRemainingMeters.value!! else STATE_UNKNOWN,
-                rangeRemaining.sensor.statelessIcon,
-                mapOf(
-                    "status" to rangeRemainingStatus
-                ),
-                forceUpdate = true
-            )
+        onSensorUpdated(
+            latestContext,
+            rangeRemaining.sensor,
+            if (rangeRemainingStatus == "success") data.rangeRemainingMeters.value!! else STATE_UNKNOWN,
+            rangeRemaining.sensor.statelessIcon,
+            mapOf(
+                "status" to rangeRemainingStatus
+            ),
+            forceUpdate = true
+        )
+        sensorWorkerScope.launch {
```

Comment:
Why are you launching something here? `setListener` is not suspending.

Reply:
ah sorry old code